### PR TITLE
Add Zig primes test

### PR DESCRIPTION
**Pull request by radszy**
This also updates docker setup and the surrounding scripts.

Signed-off-by: Radek Szymanski <radszy@pm.me>

### CODE REVIEW

File: common/commands.mk [hunk 1]
```diff
@@ -32,6 +33,7 @@ VALAC_CLANG_BUILD =	valac $^ --cc=clang -D CLANG_TEST $(VALAC_FLAGS) -o $@
 VALAC_GCC_BUILD =	valac $^ --cc=gcc -D GCC_TEST $(VALAC_FLAGS) -o $@
 V_CLANG_BUILD =	v $(V_FLAGS) -cc clang -o $@ $^
 V_GCC_BUILD =		v $(V_FLAGS) -cc gcc -o $@ $^
+ZIG_BUILD = 		zig build-exe $(ZIG_FLAGS) -lc $(LIBNOTIFY_FLAGS) -femit-bin=$@ $^
```
Comment (nuald) [inline common/commands.mk:36]: `libnotify` is rather a fall-back option, and it's preferred to use the native networking stack included to the language standard libraries (so the code could be as much self-contained as possible). As far as I see, it has `std.net.tcpConnectToAddress`, so I guess I'd prefer to use it.

File: docker/Dockerfile [hunk 2]
```diff
@@ -214,6 +214,13 @@ RUN wget --progress=dot:giga \
         && rm v_linux.zip
 ENV PATH="/opt/v/:${PATH}"
 
+# https://ziglang.org/download
+ARG ZIG=zig-linux-x86_64-0.8.0
```
Comment (nuald) [inline docker/Dockerfile:218]: It's not required to use the stable versions, it's okay to use the development versions as soon as we can track them. But it's up to you.

**Review by nuald — COMMENTED** [review]
LGTM, however if you're able to address my comments, it would be great. Thanks!

File: common/commands.mk [hunk 3]
```diff
@@ -6,6 +6,7 @@ NIM_FLAGS := -d:danger --verbosity:0 --opt:speed --hints:off --passC:"$(COMMON_F
 RUSTC_FLAGS := -C opt-level=3 -C target-cpu=native -C llvm-args=--x86-branches-within-32B-boundaries
 VALAC_FLAGS := --disable-assert -X -O3 -X -march=native -X -flto -X -Wa,-mbranches-within-32B-boundaries --pkg gio-2.0 --pkg posix
 V_FLAGS := -prod
+ZIG_FLAGS := -O ReleaseFast --single-threaded
```
[the remark this answers is not in the document]
Comment (radszy) [reply common/commands.mk:9]: This is not necessary, but it improves performance. How many other languages actually provide a feature like this? I certainly can't think of a way in C++ to make calls around mutexes a no-op using just a compiler flag.

File: common/commands.mk (hunk 1 above)
Comment (radszy) [reply common/commands.mk:36]: I'll give it a go.

File: docker/Dockerfile (hunk 2 above)
Comment (radszy) [reply docker/Dockerfile:218]: I've tried nightly version, but that wasn't making any difference. Besides, I don't know long they are keeping nightly releases for. Might be better for now to stick to stable release.

File: common/commands.mk (hunk 3 above)
[the remark this answers is not in the document]
Comment (radszy) [reply common/commands.mk]: Now removed in second commit.

File: common/commands.mk (hunk 1 above)
Comment (radszy) [reply common/commands.mk]: Now changed to std.net in second commit.

**Pull request merged**